server/src/helpers/transaction.ts

```diff
@@ -58,93 +60,150 @@ export async function getOperations(
     }
   }
 
+  var transactionFeeFromAddress = null
+  var transactionFeeAmount = null
+
+  for (const event of events) {
+    const { event: { data } } = event;
+
+    if (isTransactionFeePaidEvent(event)) {
+      transactionFeeFromAddress = data[0].toString()
+      transactionFeeAmount = (data[1] as u128)
+      
+      const transactionFeeDebitOperation = Operation.constructFromObject({
+          operation_identifier: new OperationIdentifier(operations.length),
+          type: OpType.TRANSACTION_FEE_PAID,
+          status: OperationStatus.SUCCESS,
+          account: new AccountIdentifier(transactionFeeFromAddress),
+          amount: new Amount(transactionFeeAmount.toBn().clone().neg().toString(), currency),
+      })
+
+      operations.push(transactionFeeDebitOperation);
+      break;
+    }
+  }
+
+  var transactionFeeWithdrawSkipped = false
```

Comment:
@stefanmendoza Just didn't get the point of this variable. Is it necessary?

Reply:
@osipov-mit I made `TransactionFeePaid` a balance changing event, so I want to make sure we don't double count the fee and instead skip the first withdraw that matches the amount and balance of the fee event.

https://github.com/gear-foundation/integrations-rosetta-api/pull/50/files#diff-5ca7b40e171bc3e3eccc4aea725461f6dc50334034538a7ce9fac12b57f90da5R134

Reply:
Ahh, okay, now I see